### ISSUE TITLE
[Bug] Crash everytime a menu item is clicked on: NullReferenceException

### ISSUE THREAD

**Issue by gromain** [KSP-CKAN/CKAN#3450]
Background
----------
* Operating System: Manjaro Linux
* CKAN Version: 1.30.4
* KSP Version: 1.12.1 and 1.12.2

No changes were made to any game folders.

Problem
-------
Opening a menu and clicking an item (any item) will cause an unhandled exception to occur and a subsequent crash of the executable.

This also happens when manually running `mono ckan.exe` with the latest released version manually downloaded.

**Steps to reproduce**
* Click on any menu item (any item of File, Settings or Help)
* Exception and crash

**Expected behavior**
No crash!

**CKAN error code:**
I did it three times on three different items, it looks like the error given is identical.

Click on `File > File` (to close the menu again):
```
Unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at System.Windows.Forms.Application.RunLoop (System.Boolean Modal, System.Windows.Forms.ApplicationContext context) [0x00344] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.ApplicationContext context) [0x00011] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.Form mainForm) [0x00006] in <319fd8a897144992b3f72358af09c838>:0 
  at CKAN.Main..ctor (System.String[] cmdlineArgs, CKAN.GameInstanceManager mgr, System.Boolean showConsole) [0x002fd] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main..ctor(string[],CKAN.GameInstanceManager,bool)
  at CKAN.GUI.Main_ (System.String[] args, CKAN.GameInstanceManager manager, System.Boolean showConsole) [0x0003b] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Gui (CKAN.GameInstanceManager manager, CKAN.CmdLine.GuiOptions options, System.String[] args) [0x00008] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.GameInstanceManager manager) [0x002d1] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x00251] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <933dc9d8b62a44ff9a8f866e3767654d>:0  
  ```

Click on `File> Manage Game Instances`
```
Unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at System.Windows.Forms.Application.RunLoop (System.Boolean Modal, System.Windows.Forms.ApplicationContext context) [0x00344] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.ApplicationContext context) [0x00011] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.Form mainForm) [0x00006] in <319fd8a897144992b3f72358af09c838>:0 
  at CKAN.Main..ctor (System.String[] cmdlineArgs, CKAN.GameInstanceManager mgr, System.Boolean showConsole) [0x002fd] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main..ctor(string[],CKAN.GameInstanceManager,bool)
  at CKAN.GUI.Main_ (System.String[] args, CKAN.GameInstanceManager manager, System.Boolean showConsole) [0x0003b] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Gui (CKAN.GameInstanceManager manager, CKAN.CmdLine.GuiOptions options, System.String[] args) [0x00008] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.GameInstanceManager manager) [0x002d1] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x00251] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <933dc9d8b62a44ff9a8f866e3767654d>:0  
```

Click on `File> Open Game Directory`
```
Unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at System.Windows.Forms.Application.RunLoop (System.Boolean Modal, System.Windows.Forms.ApplicationContext context) [0x00344] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.ApplicationContext context) [0x00011] in <319fd8a897144992b3f72358af09c838>:0 
  at System.Windows.Forms.Application.Run (System.Windows.Forms.Form mainForm) [0x00006] in <319fd8a897144992b3f72358af09c838>:0 
  at CKAN.Main..ctor (System.String[] cmdlineArgs, CKAN.GameInstanceManager mgr, System.Boolean showConsole) [0x002fd] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main..ctor(string[],CKAN.GameInstanceManager,bool)
  at CKAN.GUI.Main_ (System.String[] args, CKAN.GameInstanceManager manager, System.Boolean showConsole) [0x0003b] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Gui (CKAN.GameInstanceManager manager, CKAN.CmdLine.GuiOptions options, System.String[] args) [0x00008] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.GameInstanceManager manager) [0x002d1] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x00251] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <933dc9d8b62a44ff9a8f866e3767654d>:0  
```

**Comment by HebaruSan** [comment]
Since no one else has this problem and we don't control the internals of `System.Windows.Forms.Application.RunLoop`, this is probably a problem with your Mono install. Which version of Mono do you have?

**Comment by gromain** [comment]
I suspected something like that, but I think I'm up to date. The version I was running `Mono JIT compiler version 6.13.0 (main/dbe569c5d08 mer. 02 juin 2021 11:21:55 UTC)`.

I've now updated to the latest stable version from the repository to see if it changes anything, and it now works perfectly.  For the record, the version now running is `Mono JIT compiler version 6.12.0 (makepkg/c621c35ffa0 Thu Jun 17 02:48:02 PM -03 2021)`.

Closing now, thanks for the pointer in the right direction!
